app/controllers/availabilities_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class AvailabilitiesController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @time_ranges = TimeRange.order(:start_time)
+    @availabilities = load_availabilities
+    @start_time = start_time
+    @end_time = end_time
+  end
+
+  def update_availabilities
+    availability_ids = if params.key?(:availability_ids)
+                         params[:availability_ids] else []
+                       end
+    Availability.where(user_id: current_user.id).each do |availability|
+      set(availability,
+          availability_ids.include?(availability.id.to_s))
+    end
+  end
+
+  private
+
+  def set(availability, status)
+    availability.status = status ? 1 : 0
+    availability.save
+  end
+
+  def load_availabilities
+    availabilities = []
+    7.times { |day| availabilities.push(get_availabilities_day(day)) }
+    availabilities
+  end
+
+  def get_availabilities_day(day)
+    availabilities = []
+
+    @time_ranges.each do |time_range|
+      availabilities.push(get_availability(day, time_range.id))
+    end
+    availabilities
+  end
+
+  def get_availability(day, time_range_id)
+    Availability.where(user_id: current_user.id, day: day,
+                       time_range_id: time_range_id)[0] ||
```

Comment:
Use `find_by` if you only want to take one, and you're sure it'll only return one

Reply:
And by the way, if this triple is a unique tuple, you should enforce a unique constraint on this. Figure out how to add an unique index to the database